jsonschema/_utils.py

```diff
@@ -92,10 +93,10 @@ def find_additional_properties(instance, schema):
     """
 
     properties = schema.get("properties", {})
-    patterns = "|".join(schema.get("patternProperties", {}))
+    patterns = "|".join(sorted(schema.get("patternProperties", {})))
```

Comment:
Sorting ensures that this is always a single entry in the regex cache, which is quite important for performance (of this tiny function, anyway).